testing/src/main/java/com/yubico/yubikit/testing/piv/PivTestUtils.java

```diff
@@ -307,4 +416,10 @@ public static void ecKeyAgreement(PrivateKey privateKey, PublicKey publicKey) th
 
         Assert.assertArrayEquals("Secret mismatch", secret, peerSecret);
     }
+    
+    public static ManagementKeyType getManagementKeyType(PivSession piv) {
+        return piv.getVersion().isAtLeast(5, 7, 0)
+                ? ManagementKeyType.AES192
+                : ManagementKeyType.TDES;
```

Comment:
This test might return invalid value for a YubiKey where the management key type was changed. Better approach would be to use the `ManagementKeyMetadata.getKeyType()` result. Value of `ManagementKeyMetadata` can be obtained by calling `PivSession.getManagementKeyMetadata()` - note that if a YubiKey does not support `FEATURE_METADATA`, call to this method will throw an `UnsupportedOperationException`. In that case we can assume that `ManagementKeyType` is `TDES`.